src/components/select-custom/select-custom.component.ts

```diff
@@ -0,0 +1,128 @@
+import { ChangeDetectionStrategy, Component, ContentChild, ElementRef, EventEmitter, forwardRef, Input, OnChanges, OnDestroy, OnInit, Output, SimpleChanges, TemplateRef, ViewChild } from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { Subject } from 'rxjs';
+import { debounceTime, distinctUntilChanged, takeUntil } from 'rxjs/operators';
+
+@Component({
+    selector: 'ux-select-custom',
+    templateUrl: './select-custom.component.html',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [
+        {
+            provide: NG_VALUE_ACCESSOR,
+            multi: true,
+            useExisting: forwardRef(() => SelectCustomComponent)
+        }
+    ],
+    host: {
+        '[style.width.px]': 'width'
+    }
+})
+export class SelectCustomComponent<T> implements ControlValueAccessor, OnChanges, OnInit, OnDestroy {
+
+    /** Define the selected item */
+    @Input() selected: T;
+
+    /** Hide the filter input */
+    @Input() hideFilter: boolean;
+
+    /** Debounce the filter change event emitter */
+    @Input() filterChangeDebounce: number = 500;
+
+    /** Define the width of the component */
+    @Input() width: number;
+
+    /** Define the max height of the dropdown */
+    @Input() maxHeight: number;
+
+    /** Determine if we should show the loading indicator */
+    @Input() busy: boolean;
+
+    /** Define if null values are allowed */
+    @Input() allowNull: boolean;
+
+    /** Define the placeholder for the filter input */
+    @Input() placeholder: string = 'Type to filter...';
+
+    /** Emit when the selected item is changed */
+    @Output() selectedChange = new EventEmitter<T>();
+
+    /** Emit when the filter text is changed */
+    @Output() filterChange = new EventEmitter<string>();
+
+    @ContentChild('dropdownContent', { static: false }) dropdownContentRef: TemplateRef<void>;
```

Comment:
We have two templates: one might usually be bigger, but the other one is equally important. They work side-by-side and their implementations need to be connected. So they have an equal weight in the implementation. Putting both inside ng-templates stresses this "equality".

In addition, I like to put them both inside the component, so that it is clear where they belong to and are not kind of "floating" (even if that might not hinder implementing your proposal).

That's why I would like to keep it as it is.

Reply:
After our discussion from yesterday, I changed and committed it.